frontend/src/components/features/chat/messages.tsx

```diff
@@ -4,20 +4,46 @@ import { ChatMessage } from "#/components/features/chat/chat-message";
 import { ConfirmationButtons } from "#/components/shared/buttons/confirmation-buttons";
 import { ImageCarousel } from "../images/image-carousel";
 import { ExpandableMessage } from "./expandable-message";
+import { useUserConversation } from "#/hooks/query/use-user-conversation";
+import { useConversation } from "#/context/conversation-context";
+import { ConversationTrigger } from "#/api/open-hands.types";
 
 interface MessagesProps {
   messages: Message[];
   isAwaitingUserConfirmation: boolean;
 }
 
 export const Messages: React.FC<MessagesProps> = React.memo(
-  ({ messages, isAwaitingUserConfirmation }) =>
-    messages.map((message, index) => {
+  ({ messages, isAwaitingUserConfirmation }) => {
+    const { conversationId } = useConversation();
+    const { data: conversation } = useUserConversation(conversationId || null);
+
+    // Check if conversation metadata has trigger=resolver
+    const isResolverTrigger =
+      conversation?.trigger === ConversationTrigger.RESOLVER;
+
+    return messages.map((message, index) => {
       const shouldShowConfirmationButtons =
         messages.length - 1 === index &&
         message.sender === "assistant" &&
         isAwaitingUserConfirmation;
 
+      // Special case: First user message with resolver trigger
+      if (index === 0 && message.sender === "user" && isResolverTrigger) {
+        return (
+          <div key={index}>
+            <ExpandableMessage
+              type="action"
+              message={message.content}
+              customHeader="Resolver Instructions"
```

Comment:
We should use the existing header logic to make this a translatable string

Reply:
![image](https://github.com/user-attachments/assets/665b541f-5b45-4c62-991a-c3a70153e2ff)

Updated to use translatable string instead